bundles/org.corpus_tools.hexatomic.corpusedit/src/main/java/org/corpus_tools/hexatomic/corpusedit/CorpusStructureView.java

```diff
@@ -0,0 +1,464 @@
+					addMenu.setVisible(true);
+				} else {
+					// trigger a default action based on the currently selected tree item
+					StructuredSelection selected = (StructuredSelection) treeViewer.getSelection();
+					if (selected.getFirstElement() instanceof SCorpus) {
+						addDocument(toolBar.getShell());
+					} else if (selected.getFirstElement() instanceof SCorpusGraph) {
+						addCorpus(toolBar.getShell());
+					} else if (selected.getFirstElement() instanceof SDocument) {
+						// add a sibling document
+						addDocument(toolBar.getShell());
+					} else {
+						// fallback to a corpus graph, which always can be added
+						addCorpusGraph();
+					}
+				}
+			}
+		});
+
+		MenuItem addCorpusGraph = new MenuItem(addMenu, SWT.NONE);
+		addCorpusGraph.setText("Corpus Graph");
+		addCorpusGraph.setImage(ResourceManager.getPluginImage("org.corpus_tools.hexatomic.core",
+				"icons/fontawesome/project-diagram-solid.png"));
+		addCorpusGraph.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				addCorpusGraph();
+			}
+		});
+
+		MenuItem addCorpus = new MenuItem(addMenu, SWT.NONE);
+		addCorpus.setText("(Sub-) Corpus");
+		addCorpus.setImage(ResourceManager.getPluginImage("org.corpus_tools.hexatomic.core",
+				"icons/fontawesome/folder-regular.png"));
+		addCorpus.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				addCorpus(toolBar.getShell());
+			}
+		});
+
+		MenuItem addDocument = new MenuItem(addMenu, SWT.NONE);
+		addDocument.setText("Document");
+		addDocument.setImage(ResourceManager.getPluginImage("org.corpus_tools.hexatomic.core",
+				"icons/fontawesome/file-alt-regular.png"));
+		addDocument.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				addDocument(toolBar.getShell());
+			}
+		});
+
+	}
+
+	private void addCorpusGraph() {
+
+		int oldSize = projectManager.getProject().getCorpusGraphs() == null ? 0
+				: projectManager.getProject().getCorpusGraphs().size();
+
+		SCorpusGraph newGraph = projectManager.getProject().createCorpusGraph();
+		newGraph.setName("corpus_graph_" + (oldSize + 1));
+
+		selectSaltObject(newGraph, true);
+
+	}
+
+	private void addCorpus(Shell shell) {
+
+		SCorpusGraph g = null;
+		SCorpus parent = null;
+
+		// get the selected corpus graph
+		StructuredSelection selection = (StructuredSelection) treeViewer.getSelection();
+		if (selection.getFirstElement() instanceof SCorpusGraph) {
+			g = (SCorpusGraph) selection.getFirstElement();
+
+		} else if (selection.getFirstElement() instanceof SCorpus) {
+			parent = (SCorpus) selection.getFirstElement();
+			g = parent.getGraph();
+
+		} else {
+			ErrorDialog.openError(shell, "Error when adding (sub-) corpus",
+					"You can only create a (sub-) corpus when a corpus graph or another corpus is selected",
+					new Status(Status.ERROR, "unknown", null));
```

Comment:
```suggestion
					new Status(Status.ERROR, "unknown", "Constraint in the Salt data model.")); // TODO Externalize string
```

This looks better to the user than getting the error message, and then a blank `Reason:` with nothing following. String should be externalized or put in a static variable, then re-used in all the error messages in this class.